### PR TITLE
OTP 24 compatibility: extract line number from location tuple

### PR DESCRIPTION
**Pull request by tcrossland**
OTP 24 includes [changes](https://github.com/erlang/otp/pull/2664) in the Erlang compiler to include column number information. Column information obtained from the chunk data is not currently handled by `WorkspaceSymbols.build_result/4` and provokes an `ArithmeticError` exception as decribed in #503:

https://github.com/elixir-lsp/elixir-ls/blob/b9f55cc5997a590aa97357f14bfc347871b0a1bf/apps/language_server/lib/language_server/providers/workspace_symbols.ex#L480-L490

This PR changes `build_result/4` to extract the line number when a tuple `{line, column}` is read from the chunk data.

### CODE REVIEW

File: apps/language_server/lib/language_server/providers/workspace_symbols.ex [hunk 1]
```diff
@@ -477,7 +477,16 @@ defmodule ElixirLS.LanguageServer.Providers.WorkspaceSymbols do
...
+  @spec build_result(
+          key_t,
+          symbol_t,
+          String.t(),
+          nil | non_neg_integer | {non_neg_integer, non_neg_integer}
+        ) :: symbol_information_t
+  defp build_result(key, symbol, path, {line, _}) do
+    build_result(key, symbol, path, line)
+  end
+
```
Comment (lukaszsamson) [inline apps/language_server/lib/language_server/providers/workspace_symbols.ex:489]: Please add a TODO remove when we require OTP 24

Comment (tcrossland) [reply apps/language_server/lib/language_server/providers/workspace_symbols.ex:489]: I don't think it should be removed. Both patterns should be supported if we want to support OTP 24 and earlier versions.

Comment (lukaszsamson) [reply apps/language_server/lib/language_server/providers/workspace_symbols.ex:489]: In a few years eventually support for OTP < 24 will be dropped. By that time no-one will remember why there are two definitions. That's why we need a TODO

Comment (tcrossland) [reply apps/language_server/lib/language_server/providers/workspace_symbols.ex]: OTP 24 also support both [patterns](https://erlang.org/doc/man/erl_anno.html#type-location) depending on `ERL_COMPILER_OPTIONS` (see https://github.com/erlang/otp/pull/3006). I've moved the pattern matching to `build_range/1`, created a typedef `erl_location_t` and renamed `line` variables to `location`, it seems a bit cleaner.

**Review by lukaszsamson — APPROVED** [review]

**Comment by lukaszsamson** [comment]
Similar changes are needed in elixir_sense

**Review by axelson — APPROVED** [review]
Thanks! :heart:

**Pull request merged**